src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -178,6 +180,36 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return $env;
         }
 
+        if ('url' === $prefix) {
+            $parsedEnv = parse_url($env);
+
+            if (false === $parsedEnv) {
+                throw new RuntimeException(sprintf('Invalid URL in env var "%s"', $name));
+            }
+            if (!isset($parsedEnv['scheme'], $parsedEnv['host'])) {
+                throw new RuntimeException(sprintf('Invalid URL env var "%s": schema and host expected, %s given.', $name, $env));
+            }
+            $parsedEnv += [
+                'port' => null,
+                'user' => null,
+                'pass' => null,
+                'path' => null,
+                'query' => null,
+                'fragment' => null,
+            ];
+
+            // remove the '/' separator
+            $parsedEnv['path'] = '/' !== $parsedEnv['path'] ? substr($parsedEnv['path'], 1) : null;
```

Comment:
a positive ternary would be more clear IMHO :)

Reply:
indeed